api/v1beta1/temporalcluster_types.go

```diff
@@ -1002,6 +1009,9 @@ type TemporalClusterSpec struct {
 	// Image defines the temporal server docker image the cluster should use for each services.
 	// +optional
 	Image string `json:"image"`
+	// Tag defines the temporal server docker image tag the cluster should use for each services.
+	// +optional
+	Tag string `json:"tag"`
```

Comment:
IMO adding `.spec.tag` in the API can be confusing for end users.
The `.spec.version` is used as a source of truth on the whole codebase. Please remove this.

Reply:
Users of this operator may not be able to change the convention of container images. Therefore making assumption that tag will be same with version is not desired.

For example I have to use x.y version with custom repo & tag (x.y-yyyymmdd) then there is no way to do that currently. Adding tag solves the problem.

---

Another approach I considered is to introduce image struct with repo and tag fields - instead of having them flattened in the spec. We may move image related fields to it (e.g. pull policy and cred etc). I think this approach is much easier to understand but requires more changes... if you think this is better then I'll try that.

Reply:
Ok some please be more explicit on the controller behavior on the `.spec.tag` godoc.
Please explain that:
- It's an advanced fields if users wants to use custom builds
- If empty the operator will use the desired `.spec.version`

Reply:
Where would be the most appropriate place to document the controller behavior for this field?

While adding it to the Go struct field comment is an option - since it will appear in the CRD description - this s not the place where the actual logic resides. However, including it there might still be beneficial for users who rely on the CRD documentation. Thoughts? @alexandrevilain